### ISSUE TITLE
Speed up integration tests

### ISSUE THREAD

**Issue by elliotchance** [elliotchance/c2go#47]
It's nice to have one C file per function, but where there is little change in syntax between the files (like testing all the `math.h` functions) this just means the tests take a long time.

We should consider reducing these into a single test file, like `tests/math.c` which will make testing much faster.

**Comment by bongo227** [comment]
The integration tests could also probably be made to run in parallel.

**Comment by elliotchance** [comment]
Parallel would be better, but I still think that running through the entire build chain just to test if a single function outputs the correct value is proving to be overkill.

After all, these are integrations tests so they are made to test a bunch of stuff in a single round.
